pandas/core/frame.py

```diff
@@ -1909,41 +1909,71 @@ def to_dict(self, orient: str = "dict", into=dict):
             elif orient.startswith("i"):
                 orient = "index"
 
+        object_dtype_cols = {
+            col for col, dtype in self.dtypes.items() if is_object_dtype(dtype)
+        }
+        are_all_object_dtype_cols = len(object_dtype_cols) == len(self.dtypes)
         if orient == "dict":
             return into_c((k, v.to_dict(into)) for k, v in self.items())
 
         elif orient == "list":
             return into_c(
-                (k, list(map(maybe_box_native, v.tolist()))) for k, v in self.items()
+                (
+                    k,
+                    list(map(maybe_box_native, v.tolist()))
+                    if k in object_dtype_cols
+                    else v.tolist(),
+                )
+                for k, v in self.items()
             )
 
         elif orient == "split":
+            if are_all_object_dtype_cols:
```

Comment:
do we have sufficient asv's for all of these cases?

Reply:
@jreback do you want me to add asv's? I presume this means submitting a PR into main with the relevant asv's then merging that into this branch?

Reply:
I believe it's okay to update the asvs in this PR as well.